### PR TITLE
Enable x86 testing

### PR DESCRIPTION
**Pull request by Tratcher**
#949 I primarily had to use the right bitnes of dotnet.exe everywhere. ~~The only scenario I didn't work out was selfhost net461 without publishing~~ (https://github.com/aspnet/Hosting/issues/1428) - Fixed by publishing.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/DotNetCommands.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            else if (!string.IsNullOrEmpty(home))
+            {
+                result = home;
+            }
+
+            return result;
+        }
+
+        public static string GetDotNetInstallDir(RuntimeArchitecture arch)
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/DotNetCommands.cs]: Does this have to be public?

**Comment by pakrym** [comment]
Looks fine, not a fan how our code is structured but it's not a point of this PR.

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/DotNetCommands.cs (hunk 1 above)
Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/DotNetCommands.cs]: The whole class could be internal.

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs [hunk 2]
```diff
@@ -246,8 +240,14 @@ private string ModifyANCMPathInConfig(string replaceFlag, string dllName, string
...
         {
+            var programFiles = "Program Files";
+            if (DotNetCommands.IsRunningX86OnX64(DeploymentParameters.RuntimeArchitecture))
+            {
+                programFiles = "Program Files (x86)";
+            }
+
             // Get path to program files
-            var iisExpressPath = Path.Combine(Environment.GetEnvironmentVariable("SystemDrive") + "\\", "Program Files", "IIS Express", "iisexpress.exe");
+            var iisExpressPath = Path.Combine(Environment.GetEnvironmentVariable("SystemDrive") + "\\", programFiles, "IIS Express", "iisexpress.exe");
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs]: Can we use environment variable here?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs [hunk 3]
```diff
@@ -297,8 +297,24 @@ public override void Dispose()
             }
         }
 
-        // Transforms the web.config file to include the hostingModel="inprocess" element
-        // and adds the server type = Microsoft.AspNetServer.IIS such that Kestrel isn't added again in ServerTests
+        private void ModifyDotNetExePathInWebConfig()
+        {
+            // We assume the x64 dotnet.exe is on the path so we need to provide an absolute path for x86 scenarios.
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs]: Let's not assume, let's always set the full path to dotnet we think is correct to run.

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs (hunk 2 above)
Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs]: No, the environment variable changes based on the current process architecture, but we're trying to launch a process with a different arch.

**Comment by Tratcher** [comment]
(offline approval)

**Pull request merged**